pkg/cmd/step/verify/step_verify_preinstall.go

```diff
@@ -488,6 +493,33 @@ func (o *StepVerifyPreInstallOptions) verifyStorage(requirements *config.Require
 	return nil
 }
 
+func (o *StepVerifyPreInstallOptions) verifyTLS(requirements *config.RequirementsConfig) error {
+	if !requirements.Ingress.TLS.Enabled {
+		confirm := false
+		if requirements.SecretStorage == config.SecretStorageTypeVault {
+			log.Logger().Warnf("Vault is enabled and TLS is not enabled. This means your secrets will be sent to and from your cluster in the clear. We do not recommend you do this.")
+			confirm = true
+		}
+		if requirements.Webhook != config.WebhookTypeNone {
+			log.Logger().Warnf("TLS is not enabled so your webhooks will be called using HTTP. This means your webhook secret will be sent to your cluster in the clear. We do not recommend you do this.")
+			confirm = true
+		}
+		if confirm {
+			if o.BatchMode {
+				return errors.Errorf("cannot continue because TLS is not enabled.")
+			}
+			message := fmt.Sprintf("Do you wish to continue?")
+			help := fmt.Sprintf("Jenkins X needs TLS enabled to send secrets securely. We strongly recommend enabling TLS.")
```

Comment:
I think we need some docs to indicate how to enable TLS in the requirements to avoid having users fiddling around with this config. This is just a follow up on this.

Reply:
We start to cover it at https://jenkins-x.io/getting-started/boot/#ingress - what do you think is missing?